### DISCUSSION TITLE
Query HTTP server API even when Obsidian is not running?

### DISCUSSION THREAD

**Discussion by luckman212** [scambier/obsidian-omnisearch#374] [General]
I'm working on an Alfred workflow that hits the HTTP API to get results. Works great! My question is: is there any way to run this HTTP server in the background even when Obsidian is NOT running? I'd like to always be able to search, even when Obsidian app is not open.

**Comment by scambier** [comment]
Definitely not in its current form: Omnisearch is tightly coupled to Obsidian and 100% depends on it for several key features.

I've actually already thought about it when developing the HTTP API, but it would have to be a brand new application, and it'd either still depend on Obsidian to build the search index periodically, or would have to implement a good chunk of Obsidian's markdown parsing. And it would still need to run in the background to be fast enough.

**Comment by luckman212** [comment]
Thanks @scambier I appreciate that.

The issue I face is that Obsidian and/or Electron INSIST on unhiding/coming to the foreground when launching on macOS, even when launched explicitly with the `--hide` or `--background` commandline args. So there's no non-obtrusive way to launch Obsidian in the middle of my workflow.

What about allowing the HTTP server to bind to a non-private IP? Is that possible somehow? Then I could run Obsidian + Omnisearch on a background VM or something and not worry about whether it was running...

**Reply by scambier** [reply]
> What about allowing the HTTP server to bind to a non-private IP?

This isn't actually possible for security/privacy concerns, but I'll see what I can do for the next update.
